src/pks/flow/constitutive_relations/wrm/rel_perm_evaluator.cc

```diff
@@ -283,13 +291,15 @@ void RelPermEvaluator::EvaluateFieldPartialDerivative_(const Teuchos::Ptr<State>
     // -- Evaluate the model to calculate krel on cells.
     const Epetra_MultiVector& sat_c = *S->GetFieldData(sat_key_)
         ->ViewComponent("cell",false);
+    const Epetra_MultiVector& sat_gas_c = *S->GetFieldData(sat_gas_key_)
+        ->ViewComponent("cell",false);
     Epetra_MultiVector& res_c = *result->ViewComponent("cell",false);
 
     int ncells = res_c.MyLength();
     for (unsigned int c=0; c!=ncells; ++c) {
       int index = (*wrms_->first)[c];
-      res_c[0][c] = wrms_->second[index]->d_k_relative(sat_c[0][c]);
-      AMANZI_ASSERT(res_c[0][c] >= 0.);
+      res_c[0][c] = wrms_->second[index]->d_k_relative((1-beta_)*(1-sat_gas_c[0][c])+beta_*sat_c[0][c]);
```

Comment:
Can you write this out in terms of both s_g and s_l?  This should be (in pseudo-code):

```
if (wrt_key == sat_key_) {
   ...
  res_c[0][c] = d_k_relative(...) * beta}
   ...
} else if (wrt_key == sat_gas_key_) {
  ...
   res_c[0][c] = d_k_relative(..) * (beta - 1)
}
```
```